apps/extension/src/provider/InjectedAnoma.ts

```diff
@@ -0,0 +1,177 @@
+import { ChainInfo as Chain } from "@keplr-wallet/types";
+import { Anoma as IAnoma, Signer } from "@anoma/types";
+import { Anoma } from "./Anoma";
+import { Result } from "../router/types";
+
+/**
+ * Specify which methods can be proxied
+ *
+ * In the following, we proxy all methods except "version",
+ * as it can be handled locally:
+ */
+type ProxyMethod = keyof Omit<Anoma, "version">;
+
+enum ProxyRequestTypes {
+  Request = "anoma-proxy-request",
+  Response = "anoma-proxy-request-response",
+}
+
+export interface ProxyRequest {
+  type: ProxyRequestTypes;
+  id: string;
+  method: ProxyMethod;
+  args: any;
+}
+
+export interface ProxyRequestResponse {
+  type: ProxyRequestTypes.Response;
+  id: string;
+  result: Result | undefined;
+}
+
+export class InjectedAnoma implements IAnoma {
```

Comment:
Same as with `class Anoma implements IAnoma` I feel this might be better living next to the code that is only specific to the host page related class.